CodeGeneration/Sources/SyntaxSupport/Child.swift

```diff
@@ -113,20 +113,18 @@ public class Child {
 
   /// If the child ends with "token" in the kind, it's considered a token node.
   /// Grab the existing reference to that token from the global list.
-  public var tokenKind: String? {
+  public var tokenKind: Token? {
     switch kind {
-    case .token(choices: let choices, requiresLeadingSpace: _, requiresTrailingSpace: _):
+    case .token(let choices, _, _):
       if choices.count == 1 {
         switch choices.first! {
-        case .keyword: return "KeywordToken"
-        case .token(tokenKind: let tokenKind): return tokenKind
+        case .keyword: return .keyword
+        case .token(let tokenKind): return tokenKind
         }
+      } else if choices.allSatisfy(\.isKeyword) {
+        return .keyword
       } else {
-        if choices.allSatisfy({ $0.isKeyword }) {
-          return "KeywordToken"
-        } else {
-          return "Token"
-        }
+        return .unknown
```

Comment:
I agree that this is super confusing. My hope is that we can just remove it altogether (I would do it in a follow-up PR, not this one). 

The only non-trivial place that needs to be updated is `Child.type` in `SyntaxBuildableChild` and, IMHO, the confusion stems from the fact that we try to collapse multiple token kinds into a single one without giving any indication of doing so in the variable names.

I think the way to go about fixing this is to change the associated value of `SyntaxOrTokenNodeKind.token` to `[TokenChoice]`. Then we won’t need to call `Child.tokenKind` in `Child.type` anymore and could shift the computation that’s currently done in this property down to uses where there’s a lot more context. Potentially, the computation might also just become a lot simpler once we see that context.

I think the implementation is good enough for now. Maybe just add a comment along the lines

```swift
// FIXME: Technically, returning `.unknown` is not correct here. The old string-based implementation returned "Token", which no longer exists if we model the token kinds as an enum in code generation. We should try to remove this computed property altogether <link to this discussion>
```

Reply:
Thanks. 

I've added the following comment to the code:
```swift
/*
 FIXME: Technically, returning `.unknown` is not correct here.
 The old string-based implementation returned "Token" to ensure that `tokenKind` is not nil
 and that `isToken` computed-property will return true, but the value "Token" had never been
 used in other cases. We should try to remove this computed property altogether in the issue:
 https://github.com/apple/swift-syntax/issues/2010
 */
```
I made some modifications to your original proposal to provide additional context regarding the implementation - I hope this would be useful.

I also created issue #2010 to address this. I had some difficulties in crafting the title and description, so feel free to make any necessary corrections or adjustments to clarify the issue."